engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/CompilationResult.scala

```diff
@@ -76,18 +81,24 @@ object CompilationResult extends Applicative[CompilationResult] {
           logger.warn(s"Merging expression typing info for the same node ids, overlapping expression ids: ${expressionsIntersection.mkString(", ")}. " +
             s"This can be a bug in code or duplicated node ids with same expression ids")
         }
+        if (x.parameters.isDefined && y.parameters.isDefined) {
+          logger.warn(s"Merging different parameters: ${x.parameters} and ${y.parameters}. " +
+            s"This can be a bug in code or duplicated node ids with same expression ids")
+        }
       }
 
       // we should be lax here because we want to detect duplicate nodes and context can be different then
       // also process of collecting of expressionsTypingInfo is splitted for some nodes e.g. expressionsTypingInfo for
       // sink parameters is collected in ProcessCompiler but for final expression is in PartSubGraphCompiler
-      NodeTypingInfo(y.inputValidationContext, x.expressionsTypingInfo ++ y.expressionsTypingInfo)
+      NodeTypingInfo(y.inputValidationContext, x.expressionsTypingInfo ++ y.expressionsTypingInfo, y.parameters.orElse(x.parameters))
     }
   }
 
 }
 
-case class NodeTypingInfo(inputValidationContext: ValidationContext, expressionsTypingInfo: Map[String, ExpressionTypingInfo])
+case class NodeTypingInfo(inputValidationContext: ValidationContext,
+                          expressionsTypingInfo: Map[String, ExpressionTypingInfo],
+                          parameters: Option[List[Parameter]])
```

Comment:
I see that here there are only dynamic parameters, can you rename this variable to something like `dynamicParameterDefinitions`? Also maybe some comment that during validation we also evaluate how will look like definition of dynamic parameters to have initial definition when someone will display node? Because on first look it is not obvious if we have here values of parameters or definitions of both static and dynamic parameters...

Reply:
I added comment. I hope in the future we will have more consistent API and we will pass everything here?

Reply:
I agree, good point